vllm/_custom_ops.py

```diff
@@ -186,7 +186,7 @@ def cutlass_scaled_mm_dq(a: torch.Tensor, b: torch.Tensor,
     n = b.shape[1]
     out = torch.empty((m, n), dtype=out_dtype, device=a.device)
 
-    vllm_ops.cutlass_scaled_mm_dq(out, a, b, a_scales, b_scales)
+    vllm_ops.cutlass_scaled_mm_dq(out, a, b, scale_a, scale_b)
```

Comment:
Perhaps the `torch.empty()` call here is taking some time?

For marlin, it looks like the `vllm.ops` returns the output rather than updating `out` in place. Not sure if this was done deliberately

Reply:
@tlrmchlsmth

Reply:
Looks like marlin also calls torch.empty to allocate its output, just in C++ rather than python:
https://github.com/vllm-project/vllm/blob/7a64d24aad69e4d2548aa0bf528d9fe63428ab01/csrc/quantization/marlin/dense/marlin_cuda_kernel.cu#L1091-L1094

Reply:
Got it - probably unlikely that the python wrapper is much slower